File: IIS-backend/IIS-backend.Domain/Entities/Competition.cs
namespace IIS_backend.Domain.Entities;

public class Competition
{
    public long Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public DateTime DiscountValidUntil { get; set; }
    public string? AdditionalInfo { get; set; }

    public List<Day> Days { get; set; } = new();
    public List<Zone> Zones { get; set; } = new();
    public List<CompetitionCurrency> CompetitionCurrencies { get; set; } = new();
}

File: iis-frontend/src/model/competition.ts
import type { CurrencyDto } from './currency';

export interface CompetitionDayDto {
    id: number;
    date: string;
    basePrice: number;
}

export interface CompetitionZoneDto {
    id: number;
    capacity: number;
    characteristics: string;
    priceAddon: number;
}

export interface CompetitionBasicInfoDto {
    name: string;
    location: string;
    startDate: string;
    endDate: string;
    discountValidUntil: string;
    additionalInfo?: string | null;
    days: CompetitionDayDto[];
    zones: CompetitionZoneDto[];
    allowedCurrencies: CurrencyDto[];
}

export interface UpdateCompetitionDto {
    name: string;
    location: string;
    startDate: string;
    endDate: string;
    discountValidUntil: string;
    additionalInfo?: string | null;
}

export interface SetAllowedCurrenciesDto {
    currencyIds: number[];
}